### ISSUE TITLE
Label of search/join box in the room directory should be explicit on search and handle it better

### ISSUE THREAD

**Issue by AmandineLP** [vector-im/riot-web#1250]
Today: "Join a room (e.g. #foo:domain.com)
Expected: "Search for a room or join one if you know its address (e.g. #foo:domain.com)" 
(too long?)

Hitting 'Enter' if the field is used for search (i.e. the text is not formatted as an alias) should just search (or do nothing giving search happens dynamically) but shouldn't raise the "malformed ID" error

**Comment by ara4n** [comment]
I think the suggested verbiage is too wordy - surely "Search for a room by name or address (e.g. #foo:domain.com)" would be enough?  I don't think we need to spell out 'join' at all?

Agreed on the error.

**Comment by AmandineLP** [comment]
Right, given we now have peeking we can skip "join". Wfm.

**Comment by t3chguy** [comment]
New design fixes this
![image](https://user-images.githubusercontent.com/2403652/71367872-33107b00-259e-11ea-9f57-c03b6cf35ab4.png)
